### ISSUE TITLE
Panic earlier tonight

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#860]
> gerdog: CLiff - looks like the server died @ Dealing #243738: 9000 18000 Chowaha (High only)
gerdog: lingcod was going to act and couldn't. I tried to select a command and could't
gerdog: i also tried commend box and nothing
gerdog: command
🐭GamboMouse: same here.

Fix the panic that occurred tonight.

```
2022-02-13T05:13:08.810150198Z WARN  [mb2_commands::types] sub other.0: 13000, self.0: 7800
2022-02-13T05:13:08.811036839Z thread '<unnamed>' panicked at 'attempt to subtract with overflow', lib/commands/src/types.rs:330:20

2022-02-13T05:13:09.551610516Z   16:     0x55c8fbcda859 - <mb2_commands::types::ChipCount as core::ops::arith::Sub>::sub::h7cde3c8b36e7882e
2022-02-13T05:13:09.551679567Z                                at /usr/src/poker/lib/commands/src/types.rs:330:20
2022-02-13T05:13:09.576289394Z   17:     0x55c8fbbe76fa - mb2::table::Table::invalid_raise_to::h148fc36cc7c1dfd3
2022-02-13T05:13:09.576368575Z                                at /usr/src/poker/mb2/src/table.rs:1358:26
2022-02-13T05:13:09.577293617Z   18:     0x55c8fbbf8629 - mb2::table::Table::invalid_action::h05ce68e10e1b0f51
2022-02-13T05:13:09.577369328Z                                at /usr/src/poker/mb2/src/table.rs:1463:43
2022-02-13T05:13:09.577379418Z   19:     0x55c8fbbf8629 - <mb2::table::Table as mb2::table::UserActions>::set_action::hc2afcb16743f43af
2022-02-13T05:13:09.577385148Z                                at /usr/src/poker/mb2/src/table.rs:4843:20
2022-02-13T05:13:09.599708907Z   20:     0x55c8fbab15e6 - mb2::tournament::RunningEvent::process_user_request::h28b47b9ab46301a3
2022-02-13T05:13:09.599737117Z                                at /usr/src/poker/mb2/src/tournament.rs:1579:34
```
I'll look more into it tomorrow.

**Comment by ctm** [comment]
This is a "simple" case of unsigned underflow due to mb2 doing a subtraction of a larger number from a smaller number. The solution is to compare the two numbers before doing the subtraction, so I'll put in that fix "soon". However, I'm pretty sure the `total_amount` parameter came from the graphical interface, so I've added "Fix whatever caused the invalid RaiseTo value" (#861) as an issue so I don't forget to find and fix that issue as well.

Neither of these are particularly hard, but I did not sleep much last night due to getting to bed late and having a bunch of caffeine in my system due to the [Valenclimb's Day](https://www.facebook.com/events/308934624376923/) ten hour "running" event I participated in yesterday. Whee!

**Comment by ctm** [comment]
Boy, I'm tired. I spent a bunch of time looking through `table_requests` trying to find the `SetAction(RaiseOrDoNothing(...))` invocation with the parameters that killed us. I was unsuccessful, because mb2 does not log anything to `table_requests` until after the hand is over (so that even if someone could snoop the database, that person would not be able to know what's happening).

OTOH, at the time of the death it appears that only one player was using the graphical interface, so I should be able to simply bring up the state of that player's GUI at the hand to see what options he was provided with. That should allow me to both fix this issue as well as #861.

Bringing up a player's state is not actually automated though so it takes me a little while to set it up. Out of curiosity, I'll eat breakfast and then time how long it takes to bring it up. I'm guessing the ballpark is 15 minutes.

**Comment by ctm** [comment]
Fixed. Deploying now. Didn't choose to build a regressive test suite out of existing hands, but I'll keep that in mind for the future.
